frontend/packages/console-dynamic-plugin-sdk/src/app/useReduxStore.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useStore } from 'react-redux';
+import { applyMiddleware, combineReducers, createStore, compose, Store } from 'redux';
+import thunk from 'redux-thunk';
+import { baseReducers } from './redux';
+import { RootState } from './redux-types';
+import storeHandler from './storeHandler';
+
+const composeEnhancers =
+  (process.env.NODE_ENV !== 'production' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;
+
+/**
+ * `useReduxStore` will provide the store instance if present or else create one along with info if the context was present.
+ *
+ * @example
+ * ```ts
+ * function Component () {
+ *   const {store, storeContextPresent} = useReduxStore()
+ *   return ...
+ * }
+ * ```
+ */
+export const useReduxStore = (): { store: Store<RootState>; storeContextPresent: boolean } => {
+  const storeContext = useStore();
+  const [storeContextPresent, setStoreContextPresent] = React.useState(false);
+  const store = React.useMemo(() => {
+    // check if store exists and if not create it
+    if (storeContext) {
+      setStoreContextPresent(true);
+      storeHandler.setStore(storeContext);
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('Creating the SDK redux store');
+      setStoreContextPresent(false);
+      const storeInstance = createStore(
+        combineReducers<RootState>(baseReducers),
+        {},
+        composeEnhancers(applyMiddleware(thunk)),
+      );
+      storeHandler.setStore(storeInstance);
+    }
```

Comment:
How do `baseReducers` get applied if the store already exists?
We discussed two options:
1. app is responsible to apply themselves and we export them
2. AppInitSDK applies them automatically

I don't see either applied here.

Reply:
yeah as part of this PR, I separated the redux-store dependency in SDK and I have planned to do point 1 with other PR once we have some definitive reducers, in other PR https://github.com/openshift/console/pull/10479 I would need to refactor and expose `k8s` reducer then we can expose both the reducers and the app would apply these while creating the store. 

And for point 2, `AppInitSDK applies them automatically` i.e if store context exists then in `AppInitSdk` we apply them automatically. If we have point 1, do we need this? 

As to reapply we would need something like below but in SDK we won't have the context of appBaseReducer.

```
const nextReducers: ReducersMapObject = { ...appbaseReducers , core: sdkBaseReducers };

store.replaceReducer(combineReducers<RootState>(nextReducers));
```

Reply:
I think it makes sense to address point 1 in this PR itself with the current reducer we have in SDK and expose those

Reply:
@invincibleJai either way in this PR or another or down the road, I just wanted to make sure it was considered